### PR TITLE
Add tests to TAM utility function

### PR DESCRIPTION
**Pull request by manzoorwanijk**
This PR adds tests for TAM utility function `ticketsWithNewQuantity` post #2941

### CODE REVIEW

File: assets/src/domain/eventEditor/ui/ticketAssignmentsManager/utils/index.ts [hunk 1]
```diff
@@ -97,7 +97,7 @@ export const ticketsWithNewQuantity = ({
 		const nonNegativeTicketQuantity = parseInfinity(ticketIdToQuantityMap?.[ticketId], Infinity);
 
 		// if capacity is infinite or it's more than ticket quantity
-		if (isInfinite(minimumCapacity) || minimumCapacity > nonNegativeTicketQuantity) {
+		if (isInfinite(minimumCapacity) || minimumCapacity >= nonNegativeTicketQuantity) {
```
Comment (nicolad) [inline assets/src/domain/eventEditor/ui/ticketAssignmentsManager/utils/index.ts:100]: writing tests sometimes helps to spot such small details like these :+1:

File: assets/src/domain/eventEditor/ui/ticketAssignmentsManager/utils/ticketsWithNewQuantity.test.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+			entity: 'tickets',
+			existingData,
+			newData: {},
+			relation: 'datetimes',
+		});
+
+		const ticketsWithChangedQuantity = ticketsWithNewQuantity({
+			allDates,
+			allTickets,
+			existingData,
```
Comment (nicolad) [inline assets/src/domain/eventEditor/ui/ticketAssignmentsManager/utils/ticketsWithNewQuantity.test.ts:37]: if this is new relational data, should `existingData` be passed?

File: assets/src/domain/eventEditor/ui/ticketAssignmentsManager/utils/ticketsWithNewQuantity.test.ts [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+		});
+
+		expect(ticketsWithChangedQuantity).toEqual({});
+	});
+
+	it('returns an empty object when assignment is only removed', () => {
+		// lets modify existing data to create new one
+		const path = ['tickets', allTickets[0].id, 'datetimes'];
+		// ticket at allTickets[0] is related to two dates
+		// lets assign the ticket to only one date and remove others
```
Comment (nicolad) [inline assets/src/domain/eventEditor/ui/ticketAssignmentsManager/utils/ticketsWithNewQuantity.test.ts:48]: but if ticket is assigned to one date shouldn't this return this only relation?

Comment (nicolad) [reply assets/src/domain/eventEditor/ui/ticketAssignmentsManager/utils/ticketsWithNewQuantity.test.ts:48]: nvm. it's checking `ticketsWithNewQuantity`

File: assets/src/domain/eventEditor/ui/ticketAssignmentsManager/utils/ticketsWithNewQuantity.test.ts (hunk 2 above)
Comment (tn3rb) [reply assets/src/domain/eventEditor/ui/ticketAssignmentsManager/utils/ticketsWithNewQuantity.test.ts:37]: i think that's how it calculates that there are new changes... by comparing against existing

Comment (nicolad) [reply assets/src/domain/eventEditor/ui/ticketAssignmentsManager/utils/ticketsWithNewQuantity.test.ts:37]: makes sense, I dunno why confused `ticketsWithNewQuantity` with a hook and thought that is should be aware of the `existingData` :facepalm:

Comment (manzoorwanijk) [reply assets/src/domain/eventEditor/ui/ticketAssignmentsManager/utils/ticketsWithNewQuantity.test.ts:37]: Yeah the comparison is with new changes

File: assets/src/domain/eventEditor/ui/ticketAssignmentsManager/utils/ticketsWithNewQuantity.test.ts (hunk 3 above)
Comment (manzoorwanijk) [reply assets/src/domain/eventEditor/ui/ticketAssignmentsManager/utils/ticketsWithNewQuantity.test.ts:48]: Yes, and the logic is same

**Pull request merged**